### PR TITLE
CC-31953 Introduced How To guide for backend E2E testing

### PR DESCRIPTION
**Pull request by AsonUnique**
#### Overview:

- Jira ticket: https://spryker.atlassian.net/browse/CC-32158

### CODE REVIEW

File: docs/scos/dev/guidelines/testing-guidelines/executing-tests/test-the-glue-api.md [hunk 1]
```diff
@@ -0,0 +1,395 @@
+---
+title: Test the Glue API
+description: How to test the Glue API
+template: howto-guide-template
+---
+
+This document describes how to set up and run Glue Backend API end-to-end tests. All code references the `WishlistsRestApi` module example and the `Pyz` project namespace. When you set up and run the tests for a different project namespace or module, adjust the names accordingly.
```
Comment (abitskil) [inline docs/scos/dev/guidelines/testing-guidelines/executing-tests/test-the-glue-api.md:7]: * This guide explains how to set up and run Glue Backend API end-to-end tests using the WishlistsRestApi module as an example and the Pyz project namespace. Adjust the names accordingly for different project namespaces or modules.

File: docs/scos/dev/guidelines/testing-guidelines/executing-tests/test-the-glue-api.md [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+
+## Prerequisites
+
+Make sure the following prerequisites are met:
```
Comment (abitskil) [inline docs/scos/dev/guidelines/testing-guidelines/executing-tests/test-the-glue-api.md:11]: * Ensure the following prerequisites are met:

File: docs/scos/dev/guidelines/testing-guidelines/executing-tests/test-the-glue-api.md [hunk 3]
```diff
@@ -0,0 +1,395 @@
...
+
+#### 3. Execute tests
+
+Run the following command to execute the tests:
+
+```bash
+docker/sdk testing codecept run -c tests/PyzTest/Glue/Wishlists
+```
+
+Once the testing process is complete, you get the result of each individual test.
```
Comment (abitskil) [inline docs/scos/dev/guidelines/testing-guidelines/executing-tests/test-the-glue-api.md:395]: * The result of each individual test will be displayed once the testing process is complete.

File: docs/scos/dev/guidelines/testing-guidelines/executing-tests/test-the-glue-backend-api.md [hunk 4]
```diff
@@ -0,0 +1,271 @@
+---
+title: Test the Glue Backend API
+description: How to test the Glue Backend API
+template: howto-guide-template
+---
+
+This document describes how to set up and run Glue Backend API end-to-end tests. All code references the `ModuleRestApi` module example and the `Pyz` project namespace. When you set up and run the tests for a different project namespace or module, adjust the names accordingly.
```
Comment (abitskil) [inline docs/scos/dev/guidelines/testing-guidelines/executing-tests/test-the-glue-backend-api.md:7]: This guide explains how to set up and run Glue Backend API end-to-end tests using the ModuleRestApi module example and the Pyz project namespace. Adjust the names accordingly for different project namespaces or modules.

File: docs/scos/dev/guidelines/testing-guidelines/executing-tests/test-the-glue-backend-api.md [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+
+## Prerequisites
+
+Make sure the following prerequisites are met:
```
Comment (abitskil) [inline docs/scos/dev/guidelines/testing-guidelines/executing-tests/test-the-glue-backend-api.md:11]: Ensure the following prerequisites are met:

File: docs/scos/dev/guidelines/testing-guidelines/executing-tests/test-the-glue-backend-api.md [hunk 6]
```diff
@@ -0,0 +1,271 @@
...
+```
+
+### 2. Prepare a codeception test files
+
+#### 1. Create the `codeception.yml` with configuration required for your E2E test:
+
+**tests/PyzTest/Glue/ModuleBackend/codeception.yml**
+
+```yaml
+namespace: PyzTest\Glue\ModuleBackend
```
Comment (abitskil) [inline docs/scos/dev/guidelines/testing-guidelines/executing-tests/test-the-glue-backend-api.md:59]: * Let's change ModuleBackend => ModuleBackendApi in the document.

File: docs/scos/dev/guidelines/testing-guidelines/executing-tests/test-the-glue-backend-api.md [hunk 7]
```diff
@@ -0,0 +1,271 @@
...
+
+#### 5. Run the test
+
+Run the following command to execute the test:
+
+```bash
+docker/sdk testing codecept run -c tests/PyzTest/Glue/ModuleBackend
+```
+
+Once the testing process is complete, you get the result of each individual test.
```
Comment (abitskil) [inline docs/scos/dev/guidelines/testing-guidelines/executing-tests/test-the-glue-backend-api.md:271]: * The result of each individual test will be displayed once the testing process is complete.

File: docs/scos/dev/guidelines/testing-guidelines/executing-tests/test-the-glue-backend-api.md (hunk 6 above)
Comment (AsonUnique) [reply docs/scos/dev/guidelines/testing-guidelines/executing-tests/test-the-glue-backend-api.md:59]: Discussed, and decided to leave it as is and adjust once we have conventions for backend API tests.

**Review by abitskil — APPROVED** [review]

File: docs/scos/dev/guidelines/testing-guidelines/test-helpers/available-test-helpers.md [hunk 8]
```diff
@@ -232,6 +222,35 @@ Calling in your test `$this->tester->setDependency(OmsDependencyProvider::FACADE
...
+
+Lets you set dependencies to a module or mock existing ones.
+
+#### FactoryHelper
+
+Lets you mock factory methods and access the factory.
+
+#### OpenApi3
+
+Lets you match the response of API endpoints against the OpenApi3 schema.
```
Comment (andriitserkovnyi) [inline docs/scos/dev/guidelines/testing-guidelines/test-helpers/available-test-helpers.md:239]: Can we replace `match` with `validate`?

File: docs/scos/dev/guidelines/testing-guidelines/executing-tests/test-the-glue-api.md [hunk 9]
```diff
@@ -0,0 +1,424 @@
+---
+title: Test the Glue API
+description: Learn how to test the Glue API
+template: howto-guide-template
+---
+
+This guide explains how to set up and run Glue API end-to-end (E2E) tests using the `WishlistsRestApi` module as an example and the `Pyz` project namespace. Adjust the names according to your project namespaces and modules.
```
Comment (andriitserkovnyi) [inline docs/scos/dev/guidelines/testing-guidelines/executing-tests/test-the-glue-api.md:7]: Are we using the project namespace as an example as well?

File: docs/scos/dev/guidelines/testing-guidelines/test-helpers/available-test-helpers.md [hunk 10]
```diff
@@ -232,6 +222,35 @@ Calling in your test `$this->tester->setDependency(OmsDependencyProvider::FACADE
...
+
+Lets you navigate through the JSON response of the API endpoints.
+
+#### GlueRest
+
+Provides functionality to send requests to the Glue API endpoints and validate response data.
+
+#### GlueBackendApiJsonApiHelper
+
+Provides functionality to send requests to the Glue Backend API endpoints that implement JSON API convention and validate response data.
```
Comment (andriitserkovnyi) [inline docs/scos/dev/guidelines/testing-guidelines/test-helpers/available-test-helpers.md:251]:
```suggestion
Lets you send requests to the Glue Backend API endpoints that implement JSON API convention and validate response data.
```

**Pull request merged**